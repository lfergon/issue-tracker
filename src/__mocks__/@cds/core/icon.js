module.exports = {
  ClarityIcons: {
    addIcons: jest.fn()
  }
};