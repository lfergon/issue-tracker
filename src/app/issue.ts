export interface Issue {
  issueNumber: number;
  title: string;
  description: string;
  priority: 'low' | 'medium' | 'high';
  type: 'bug' | 'feature' | 'task';
  completedAt?: Date;
}
