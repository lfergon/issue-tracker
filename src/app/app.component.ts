import { Component } from '@angular/core';
import { IssueListComponent } from './issue-list/issue-list.component';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
  imports: [IssueListComponent],
})
export class AppComponent {
  title = 'issue-tracker';
}
