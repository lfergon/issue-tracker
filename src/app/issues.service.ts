import { Injectable } from '@angular/core';
import { Issue } from "./issue";
import { issues } from "../assets/mock-issues";

@Injectable({
  providedIn: 'root'
})
export class IssuesService {
  private issues: Issue[] = issues;

  constructor() { }

  getPendingIssues(): Issue[] {
    return this.issues.filter(issue => !issue.completedAt);
  }

  createIssue(issue: Issue): void {
    issue.issueNumber = this.issues.length + 1;
    this.issues.push(issue);
  }

  completeIssue(issue: Issue): void {
    const selectedIsssue: Issue = {
      ...issue,
      completedAt: new Date()
    };
    const index = this.issues.findIndex(issue => issue.issueNumber === selectedIsssue.issueNumber);
    this.issues[index] = selectedIsssue;
  }

  getSuggestions(title: string): Issue[] {
    if (title.length > 3) {
      return this.issues.filter(issue => issue.title.includes(title));
    }
    return [];
  }
}
