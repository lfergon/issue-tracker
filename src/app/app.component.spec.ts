import { ComponentFixture, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let app: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AppComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(AppComponent);
    app = fixture.componentInstance;
  });

  it('should create the app', () => {
    expect(app).toBeTruthy();
  });

  it(`should have as title 'issue-tracker'`, () => {
    expect(app.title).toEqual('issue-tracker');
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
