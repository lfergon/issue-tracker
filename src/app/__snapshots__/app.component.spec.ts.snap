// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<app-root
  title={[Function String]}
>
  <div
    class="main-container"
  >
    <div
      class="content-container"
    >
      <div
        class="content-area"
      >
        <app-issue-list />
      </div>
    </div>
  </div>
</app-root>
`;
